flutter_modular/lib/src/presenters/navigation/modular_route_information_parser.dart

```diff
@@ -191,17 +191,37 @@ class ModularRouteInformationParser extends RouteInformationParser<ModularRoute>
     if (path.isEmpty) {
       throw Exception("Router can not be empty");
     }
+    var uri = Uri.tryParse(path);
+    Map<String, List<String>>? queryParams;
+    String? fragment;
+    // the fragments and query params are removed from the path so that it doesn't affect route matching etc
+    //i.e /dashboard should also match /dashboard?id=12
+    if (uri != null) {
+      queryParams = uri.queryParametersAll;
```

Comment:
Wouldn't it be better to use the "queryParameters" property instead of "queryParametersAll"?

Reply:
We could return `<String,String>` but as mentioned above, params such as `http://example.com?ids=23&ids=24&ids=25` will take only the first value and the others will be lost. Now, while this might seem easier, I was trying to avoid someone creating an issue or raising a support request later. 

Since the support for queryParams is being added, might as well support it fully.

Let me know if you still want me to change it from `<String,List<String>>` to `<String,String>`

Reply:
Got it!
In that case, I believe it would be prudent to have both. However, I also thought that we could store the entire URI and create getters for the path and queryParams from the URI.

Reply:
I believe you're right. I will add functionality after diving into your PR. Thanks!